lang/cs/Org.Apache.REEF.Client/Common/JobSubmissionResultImpl.cs

```diff
@@ -120,29 +150,28 @@ internal static string GetTrackingUrl(string filePath)
             return result.Item2;
         }
 
-        internal string GetDriverUrlForYarn(String filePath)
+        internal string GetDriverUrlForYarn(string filePath)
         {
-            _driverUrl = GetTrackingUrl(filePath);
-            return _driverUrl;
+            return GetTrackingUrlAppId(filePath);
         }
 
         internal string GetDriverUrlForLocalRuntime(string filePath)
         {
-            _driverUrl = null;
+            string fullDriverUrl = null;
             for (int i = 0; i < 10; i++)
             {
                 var driverUrl = TryReadHttpServerIpAndPortFromFile(filePath);
                 if (!string.IsNullOrEmpty(driverUrl))
                 {
-                    _driverUrl = "http://" + driverUrl + "/";
+                    fullDriverUrl = "http://" + driverUrl + "/";
                     break;
                 }
                 Thread.Sleep(1000);
```

Comment:
In this case, we are waiting for the data to be written to the file. If it is not ready, we need to wait for sometime and then try again. We don't want to keep trying and using CPU time.